charms/kfp-profile-controller/tests/unit/test_operator.py

```diff
@@ -164,7 +167,11 @@ def test_install_with_all_inputs(harness, oci_resource_data):
     harness.add_relation_unit(os_rel_id, "storage-provider/0")
     harness.update_relation_data(os_rel_id, "storage-provider", os_data)
 
-    harness.begin_with_initial_hooks()
+    pod_spec = harness.get_pod_spec()
```

Comment:
Since you create the `pod_spec` here, we can reuse `pod_spec` in the assertion a few lines after this rather than invoke it again.